packages/client/src/pages/CheckerPage.tsx

```diff
@@ -1,313 +1,336 @@
+      );
+    }
+  };
 
   // In case `topicData` is not found on the Session Context display an error
   // This is to prevent a bug when `topicData` is passing old data or when the Session Storage is manually deleted
   if (!sessionContext.session[slug]) {
     return <ErrorPage />;
   }
 
-  /**
-   * Set the questionIndex the next questionId, previous questionId, or the given id.
-   */
-  const goToQuestion = (value: GoToQuestionProp) => {
-    let action, eventName;
-    let newQuestionIndex: number;
-
-    if (value === "next" || value === "prev") {
-      // Either go 1 question next or prev
-      newQuestionIndex =
-        value === "next" ? questionIndex + 1 : questionIndex - 1;
-
-      if (!checker.stack[newQuestionIndex]) {
-        captureException(
-          `Go to question, question with index: ${newQuestionIndex} not found on stack`
-        );
-        return;
-      }
+  const sectionFunctions: SectionFunctions = {
+    activateSection,
+    completeSection,
+    getNextSection,
+    goToNextSection,
+    goToPrevSection,
+  };
 
-      // Matomo event props
-      action = checker.stack[questionIndex].text;
-      eventName =
-        value === "next"
-          ? eventNames.GOTO_NEXT_QUESTION
-          : eventNames.GOTO_PREV_QUESTION;
-    } else {
-      // Edit specific question index (value), go directly to this new question index
-      newQuestionIndex = value;
-
-      if (!checker.stack[newQuestionIndex]) {
-        captureException(
-          `Go to question, question with index: ${newQuestionIndex} not found on stack`
-        );
-        return;
+  // @TODO: refactor this style
+  const activeStyle = { marginTop: -1, borderColor: "white" };
+
+  const sectionsRenderer = () =>
+    sections.map((section, computedIndex) => {
+      const {
+        isActive,
+        isCompleted,
+        component,
+        heading,
+        hideSection,
+        renderOutsideWrapper,
+      } = section;
+
+      if (hideSection) {
+        // Skip to the next section if the section to be hidden is active
+        if (isActive) {
+          goToNextSection();
+        }
+        return null;
       }
 
-      // Matomo event props
-      action = actions.EDIT_QUESTION;
-      eventName = (checker.stack[newQuestionIndex] as any).text;
-    }
-
-    matomoTrackEvent({
-      action,
-      name: eventName,
-    });
-
-    setTopicData({
-      questionIndex: newQuestionIndex,
-    });
-  };
+      const componentToRender =
+        component &&
+        component({
+          currentSection: section,
+          sectionFunctions,
+        });
 
-  // Callback to go to the Outcome section
-  // `false` is to prevent unexpected click, hover and focus states on already active section
-  const handleOutcomeClick =
-    !isActive(sections.OUTCOME) && isFinished(sections.QUESTIONS)
-      ? () => setActiveState(sections.OUTCOME)
-      : false;
+      const handleOnClick = () => {
```

Comment:
Can't we just call activeSection directly?

Reply:
Thanks. I've updated this